sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CoderRegistryTest.java

```diff
@@ -89,8 +90,14 @@ public void testSerializableFallbackCoderProvider() throws Exception {
   public void testProtoCoderFallbackCoderProvider() throws Exception {
     CoderRegistry registry = getStandardRegistry();
 
+    //must use reflection here or a reference to dataflow class is baked into
```

Comment:
> However, when the test is compiled with edt, the byte code is different enough that calls to the CoderRegistryTest.class.getName() and several other things that that are done internally will throw an exception as it tries to completely validate the entire class and the dataflow class isn't available.

I can't repro this. Here's what I see after reverting `CoderRegisterTest.java` changes and running `mvn -Peclipse-jdt clean integration-test`:

1. With current PR, extra tests get run during integration test phase; `CoderRegistryTest.testIgnoredGenericInPipeline` and `CoderRegistryTest.testSpecializedButIgnoredGenericInPipeline` fail with stack trace you posted above.
2. When bumping `tycho-compiler-jdt` artifact version to 0.26 in the pom.xml, only integration tests are run and the run succeeds.

I think bumping the tycho version and reverting these test changes should be sufficient. If you're still seeing other issues, can you paste more details?